CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -400,6 +393,64 @@ local function CreateMenuItem(origInstance)
 	return this
 end
 
+local function Create3DMenuItem()
+	local menuItem = {}
+
+	local backgroundButton = Util.Create 'ImageButton' {
+		Name = "ButtonBackground",
+		Size = UDim2.new(0, 50, 0, 50),
+
+		BackgroundTransparency = 1
+	}
+	
+
+	--Unfortunately we need a clone of the selection ring so we can hide
+	--the ring if this button is active. hm.
+	backgroundButton.SelectionImageObject = selectionRing:Clone()
+
+	local isActive = false
+
+	function menuItem:SetActive(active)
+		isActive = active
+		if active then
+			backgroundButton.Image = "rbxasset://textures/ui/menu/buttonActive.png"
+		else
+			backgroundButton.Image = "rbxasset://textures/ui/menu/buttonBackground.png"
+		end
+		backgroundButton.SelectionImageObject.Visible = not active
+	end
+
+	function menuItem:GetWidth()
+		return backgroundButton.AbsoluteSize.X
+	end
+
+	function menuItem:GetInstance()
+		return backgroundButton
+	end
+
+	function menuItem:SetTransparency(transparency)
+		backgroundButton.ImageTransparency = transparency
+		for i, v in pairs(backgroundButton:GetChildren()) do
```

Comment:
Is it possible the we will need to recursively call this on the children's children at some point?